packages/next/src/server/lib/router-server.ts

```diff
@@ -583,13 +583,19 @@ export async function initialize(opts: {
       })
 
       if (opts.dev && developmentBundler && req.url) {
-        const isHMRRequest = req.url.includes('/_next/webpack-hmr')
+        const { basePath } = config
+
+        const isHMRRequestForHostApp =
+          !basePath && req.url.startsWith('/_next/webpack-hmr')
+
+        const isHMRRequestForGuestApp = pathHasPrefix(req.url, basePath)
+
+        const isHMRRequestForCurrentBasepath =
```

Comment:
With this being a solution for the HMR multizone, is this line correct?

Given zone 1 is called from zone 2 during hmr, at least one of those Boolean values would always return true, which would mean line 598 would always be true

Reply:
Would you not want something like:
` isHMRRequestForHostApp && !isHMRRequestForGuestApp`

Reply:
I tried your solution locally, and the infinite refresh still happened 
as` isHMRRequestForHostApp || isHMRRequestForGuestApp`, one of those values would always be true, i changed it to:
` isHMRRequestForHostApp && !isHMRRequestForGuestApp` and this worked.